File: src/main/java/io/shiftleft/fuzzyc2cpg/parser/CodeLocationExtractor.java
package io.shiftleft.fuzzyc2cpg.parser;

import io.shiftleft.fuzzyc2cpg.ast.CodeLocation;
import org.antlr.v4.runtime.ParserRuleContext;

public class CodeLocationExtractor {

  public static CodeLocation extractFromContext(ParserRuleContext ctx) {
    CodeLocation location = new CodeLocation();

    location.startLine = ctx.start.getLine();
    location.startPos = ctx.start.getCharPositionInLine();
    location.startIndex = ctx.start.getStartIndex();
    if (ctx.stop != null) {
      location.stopIndex = ctx.stop.getStopIndex();
    } else {
      location.stopIndex = CodeLocation.NOT_SET;
    }

    return location;
  }
}

File: src/main/scala/io/shiftleft/fuzzyc2cpg/parser/CodeLocationExtractor.scala
package io.shiftleft.fuzzyc2cpg.parser

import io.shiftleft.fuzzyc2cpg.ast.CodeLocation
import org.antlr.v4.runtime.ParserRuleContext

object CodeLocationExtractor {

  def extractFromContext(ctx: ParserRuleContext): CodeLocation = {
    val startLine = Some(ctx.start.getLine)
    val startPos = Some(ctx.start.getCharPositionInLine)
    val startIndex = Some(ctx.start.getStartIndex)
    val endLine = if (ctx.stop != null) { Some(ctx.stop.getLine) } else { None }
    val endIndex = if (ctx.stop != null) { Some(ctx.stop.getStopIndex) } else { None }
    val endPos = if (ctx.stop != null) { Some(ctx.stop.getCharPositionInLine) } else { None }
    new CodeLocation(startLine, startPos, startIndex, endIndex, endLine, endPos)
  }

}
